telethon_examples/auto_reply.py

```diff
@@ -0,0 +1,115 @@
+                'api_hash':environ['TG_API_HASH'],
+                'user_phone':environ['TG_PHONE'],
+                'session_name':'needsmore'}
+        super().__init__(
+            settings.get('session_name','session1'),
+            settings['api_id'],
+            settings['api_hash'],
+            proxy=None,
+            process_updates=True)
+
+        user_phone = settings['user_phone']
+
+        print('INFO: Connecting to Telegram Servers...', end='', flush=True)
+        self.connect()
+        print('Done!')
+
+        if not self.is_user_authorized():
+            print('INFO: Unauthorized user')
+            self.send_code_request(user_phone)
+            code_ok = False
+            while not code_ok:
+                code = input('Enter the auth code: ')
+                try:
+                    code_ok = self.sign_in(user_phone, code)
+                except RPCError as e:
+                    if e.password_required:
+                        pw = getpass('Two step verification enabled. Please enter your password: ')
+                        self.sign_in(password=pw)
+                    else:
+                        raise e
+            print('INFO: Client initialized succesfully!')
+
+    def run(self):
+        # Listen for updates
+        while True:
+            update = self.updates.poll() # This will block until an update is available
+            triggers = []
+            if isinstance(update, Updates):
+                for x in update.updates:
+                    if not isinstance(x,UpdateNewChannelMessage): continue
+                    if isinstance(x.message,MessageService): continue
+                    # We're only interested in messages to supergroups
+                    words = word_tokenize(x.message.message.lower())
+                    # Avoid matching 'emacs' in 'spacemacs' and similar
+                    if 'emacs' in words and x.message.to_id.channel_id not in EMACS_BLACKLIST:
+                        triggers.append(('emacs',x.message))
+                    if 'chrome' in words:
+                        triggers.append(('chrome',x.message))
+                    if 'x files theme' == x.message.message.lower() and x.message.out:
+                        # Automatically reply to yourself saying 'x files theme' with the audio
+                        msg = x.message
+                        chan = InputPeerChannel(msg.to_id.channel_id,CHANNELS[msg.to_id.channel_id])
+                        self.send_voice_note(chan,'xfiles.m4a',reply_to=msg.id)
+                        sleep(1)
+                    if '.shrug' in x.message.message.lower() and x.message.out:
+                        # Automatically replace '.shrug' in any message you
+                        # send to a supergroup with the shrug emoticon
+                        msg = x.message
+                        chan = InputPeerChannel(msg.to_id.channel_id,CHANNELS[msg.to_id.channel_id])
```

Comment:
The channel (or supergroup) will be cast to InputPeerChannel for you, no need to perform this conversion.

Reply:
The message to_id only gives me the channel id, the access hash is taken from a file. I don't have a peer to cast unless I make one by hand.